src/App.js

```diff
@@ -3,16 +3,28 @@ import 'font-awesome/css/font-awesome.min.css';
 import Header from "./components/Header/Header";
 import Footer from "./components/Footer/Footer";
 import Main from "./components/Main/Main";
+import { useState } from "react";
 
-function App() {
+const App = () => {
+  const [showHomePage, setShowHomePage] = useState(true);
+  const [showAllProducts, setShowAllProducts] = useState(false);
+
+  const onViewChange = (showHomePage,showAllProducts) => {
+    setShowAllProducts(showAllProducts);
+    setShowHomePage(showHomePage);
+  };
```

Comment:
You can change this, right now, there is only two pages, so you can reduce it for `isHomePage`

```tsx
// The conditional render should be something like this

isHomePage ? <HomePage ..... /> : <ProductList ... />

```